tests/multi_gcov_aggregator.py

```diff
@@ -0,0 +1,145 @@
+# Script to merge coverage files generated by gcov
+# Author: Noe Brucy, Tine Colman
+# Known issues:
+# - in rt_spectra, gcov identified some variable declarations as executed.
+# - in turb_force_utils, stop is identified as a branch
+
+import os
+import re
+from collections import defaultdict
+from glob import glob
+
+class GCovParser:
+    def __init__(self):
+        # Dictionary to store data as {source_file: {line_number: (execution_count, line_content)}}
+        self.coverage_data = defaultdict(lambda: defaultdict(lambda: ("-", "", [])))
+
+    def parse_gcov_file(self, file_path, directory=None):
+        """
+        Parse a single GCOV file and extract coverage data.
+        """
+        source_file = None
+        with open(file_path, 'r') as file:
+            for line in file:
+                # Identify source file
+                if line.startswith("        -:    0:Source:"):
+                    source_file = line.split("Source:")[1].strip()
+                    continue
+
+                if source_file is None:
+                    continue  # Skip until the source file is identified
+
+                # Match GCOV execution counts: execution_count: line_number: line_content
+                match = re.match(r'^\s*(\-|#####|\d+\*?|\d+):\s*(\d+):\s*(.*)', line)
+                if match:
```

Comment:
For the sake of readability, I'd suggest:
```suggestion
                if not match:
                    continue
```
and deindent the rest :)